### PR TITLE
Add TWTSideMenuViewControllerDelegate for will/did open and close

### PR DESCRIPTION
**Pull request by armintelker**
Hey Joshm 
i added at TWTSideMenuViewController  a TWTSideMenuViewControllerDelegate because i need to listen when the menu will and did close and open.

Best regards from Germany
Armin Telker

### CODE REVIEW

**Comment by jnjosh** [comment]
Thanks for making this pull request! I have a couple changes I'd like before merging it in, I'll comment on the individual lines.

File: TWTSideMenuViewController/TWTSideMenuViewController.h [hunk 1]
```diff
@@ -24,6 +24,14 @@
 
 @class TWTSideMenuViewController;
 
+@protocol TWTSideMenuViewControllerDelegate <NSObject>
+@optional
+- (void)willOpenMenu:(TWTSideMenuViewController *)sender;
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.h]: I'd prefer if these methods were named a little better. Follow table view's delegates for inspiration:
- (void)sideMenuViewControllerWillOpen:(TWTSideMenuViewController *)sideMenuViewController;

etc.

File: TWTSideMenuViewController/TWTSideMenuViewController.h [hunk 2]
```diff
@@ -31,6 +39,9 @@ typedef NS_ENUM(NSInteger, TWTSideMenuAnimationType) {
     TWTSideMenuAnimationTypeFadeIn //New View controllers fade in over the old view controller.
 };
 
+/** The TWTSideMenuViewControllerDelegate */
+@property (nonatomic,strong) id delegate;
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.h]: As this is a delegate this should be weak. Also, it should require an object conforming to TWTSideMenuViewControllerDelegate.

@property (nonatomic, weak) id&lt;TWTSideMenuViewControllerDelegate&gt; delegate;

File: TWTSideMenuViewController/TWTSideMenuViewController.h [hunk 3]
```diff
@@ -31,6 +39,9 @@ typedef NS_ENUM(NSInteger, TWTSideMenuAnimationType) {
     TWTSideMenuAnimationTypeFadeIn //New View controllers fade in over the old view controller.
 };
 
+/** The TWTSideMenuViewControllerDelegate */
```
Comment (jnjosh) [inline TWTSideMenuViewController/TWTSideMenuViewController.h]: This comment doesn't say much more than what the property itself should describe. Try explaining more about what it expects and maybe discuss the protocol it must conform to.

**Comment by armintelker** [comment]
Hey Josh, i tried to refactor the code to get them ready for a merge into the master.
Best regards :)
Armin Telker

**Comment by jnjosh** [comment]
:+1:

**Pull request merged**